File: system-test/sec-tests/java/Sec43Observability.java
// SEC-43: Observability Data Leak
// Test: TC-CR-047-B
package com.example;

import org.springframework.web.bind.annotation.*;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.core.env.Environment;

@RestController
public class ActuatorController {

    @Autowired
    private Environment env;

    @GetMapping("/actuator/env")
    public String getEnv() {
        // Exposes all environment variables including SECRET_KEY, passwords
        return env.toString();
    }
}

File: system-test/sec-tests/python/sec_43_observability.py
# SEC-43: Observability Data Leak
# Test: TC-CR-047-B
import os
from flask import Flask, jsonify

app = Flask(__name__)

@app.route('/actuator')
def actuator():
    # Exposes all internal endpoints including env vars
    return jsonify(health=check_health(), env=dict(os.environ))
